beacon-chain/rpc/eth/helpers/lightclient/light_client.go

```diff
@@ -0,0 +1,371 @@
+package lightclient
+
+import (
+	"bytes"
+	"context"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/prysmaticlabs/prysm/v3/beacon-chain/state"
+	"github.com/prysmaticlabs/prysm/v3/config/params"
+	"github.com/prysmaticlabs/prysm/v3/consensus-types/interfaces"
+	types "github.com/prysmaticlabs/prysm/v3/consensus-types/primitives"
+	"github.com/prysmaticlabs/prysm/v3/encoding/bytesutil"
+	ethpbv1 "github.com/prysmaticlabs/prysm/v3/proto/eth/v1"
+	ethpbv2 "github.com/prysmaticlabs/prysm/v3/proto/eth/v2"
+	"github.com/prysmaticlabs/prysm/v3/proto/migration"
+	"github.com/prysmaticlabs/prysm/v3/time/slots"
+)
+
```

Comment:
This file implements the ETH2 light client sync protocol helper functions for the full node part.

Reply:
could these go in proto/eth/service/v2/custom.go instead? Also maybe call them `New` instead of `Create`?

Reply:
Sure, we can move it anywhere.   
Not sure `proto/eth/service/v2/custom.go` is a good place.    I thought `proto` folder might be used for protobuf file and generated code only?

Reply:
* Tried moving this to 
    proto/eth/service/v2 
which will require many refactoring (to solve cycle dependencies)

     * Current impl for these spec defined helper functions is using `migration` pkg
     * `migration` pkg has a dependency of `ethpbv2 (proto/eth/service/v2)`
     
     * So if moving these to `ethpbv2` pkg, then it will have 
           ethpbv2 depends on migration depends ethpbv2  
        The solution is to move those common code into a different pkg.   (Just like current doing)
        
So let's keep the current pkg structure?

Reply:
Sounds good, definitely not worth refactoring if it introduces cyclic deps. Re using `custom.go` in that package I had no choice for the other methods as they are methods on the objects themselves (`FinalityUpdate` etc) to implement the `LightClientUpdate` interface. Your new methods are also constructors for those types so they could fit there if it wasn't for the issue above.